src/components/MemoryCard.vue

```diff
@@ -1,49 +1,89 @@
 <script lang="ts">
-import { PropType } from "vue";
-import { CardContent } from "../types/DataModels";
-export default {
+import { defineComponent, PropType } from "vue";
+import { CardData, CardType } from "../types/DataModels";
+export default defineComponent({
   props: {
     cardContent: {
-      type: Object as PropType<CardContent>,
+      type: Object as PropType<CardData>,
       required: true,
     },
+    index: Number,
+    canTurn: Boolean,
+    initiallyRevealed: Boolean,
+  },
+  data() {
+    return {
+      turnedOver: false,
+      cardRotation: 180,
+      revealedClass: {
+        transform: "rotate(180deg)",
+      },
+    };
+  },
+  watch: {
+    canTurn(newValue: boolean) {
+      if (newValue == true && this.turnedOver) {
+        this.flipCard();
+      }
+    },
   },
   mounted(this: {
     setContent: () => void;
+    flipCard: () => void;
     $refs: { text: HTMLParagraphElement; image: HTMLImageElement };
+    initiallyRevealed: boolean;
   }) {
     this.setContent();
+    if (this.initiallyRevealed == true) {
+      this.flipCard();
+    }
   },
   methods: {
+    revealCard(this: any) {
+      if (
+        (this as { $refs: { card: HTMLElement } }).$refs.card &&
+        this.canTurn &&
+        !this.turnedOver
+      ) {
+        (this as { $refs: { card: HTMLElement } }).$refs.card.classList.toggle(
```

Comment:
I am not sure if this is the cleanest way to solve this. You could introduce a new variable in CardDate wheter it is flipped or not and make the MemoryCard class dynamic. e.g.:
```
<MemoryCard
          :cardContent="card"
          :index="index"
          :canTurn="canTurnCard"
          v-show="!card.found"
          @cardToggle="cardClickProcedure"
	        :class="{
      		    'flipped': card.flipped,
    	    }"
/>
```